### ISSUE TITLE
move more functions from scrapy.utils.url to w3lib

### ISSUE THREAD

**Issue by kmike** [scrapy/scrapy#2186]
Hey,

What do you think about moving `add_http_if_no_scheme` and `escape_ajax` functions to w3lib?

**Comment by redapple** [comment]
Good idea for `escape_ajax`, looks generic and useful enough. (Should we also have an `unescape_ajax`?)
I'm not sure about `add_http_if_no_scheme()`. Would it be useful outside of scrapy? Also one couldargue why it adds "http://" and not "https://".

**Comment by kmike** [comment]
I'm using `add_http_if_no_scheme` in pretty much every project where user provides an URL as an input. I think it should add http, not https because this is what browsers add if you enter an URL to address bar without a schema. Many websites don't work with https, so adding https to a user-provided URL is likely to break it; sites which work with https often provide a redirect from http to https version anyways.

**Comment by kmike** [comment]
hm, what's the use case for unescape_ajax?

**Comment by redapple** [comment]
> hm, what's the use case for unescape_ajax?
I was thinking to make ajax-escaped URLs more readable. Is `_escaped_fragment_` still that commonly used anyway?

**Comment by redapple** [comment]
Also, we have a couple of dual functions in `w3lib.url` already (`add_or_replace_parameter`/`url_query_parameter`, `file_uri_to_path`/`path_to_file_uri`)

**Comment by kmike** [comment]
I think Google deprecated `_escaped_fragment_`, but as there are sites who support it escape_ajax can still be useful for crawling.

**Comment by kmike** [comment]
unescape_ajax looks less useful because there are no ajax-escaped links on websites. It may be helpful to parse logs though.

**Comment by patiences** [comment]
Hi! I'd love to give this a shot -- if it's still relevant? :-)
